### PR TITLE
Catch By Reference

### PR DESCRIPTION
**Pull request by ax3l**
C++ catch parameters should be passed by const reference.

Sorry for the cross-over change.

Did not see relevant implications right now, but should be fixed.

References:
  - http://en.cppreference.com/w/cpp/language/try_catch

### CODE REVIEW

**Comment by PrometheusPi** [comment]
@ax3l You are author of the commit but did not update the license header.
Did you want to use `tool` as author or did you miss updating the header?

File: src/tools/splash2txt/splash2txt.cpp [hunk 1]
```diff
@@ -159,8 +159,9 @@ bool parseOptions( int argc, char** argv, ProgramOptions &options )
         }
 
     }
-    catch ( boost::program_options::error e )
+    catch ( const boost::program_options::error& )
     {
+        errorStream << desc << "\n";
```
Comment (PrometheusPi) [inline src/tools/splash2txt/splash2txt.cpp]: why did you add a error stream output here?
(not mentioned in the pull request description)

**Comment by PrometheusPi** [comment]
@ax3l great job

File: src/tools/splash2txt/splash2txt.cpp (hunk 1 above)
Comment (ax3l) [inline src/tools/splash2txt/splash2txt.cpp]: oh I forgot that I added that. it's just printing "--help" before throwing the error that something in the input was parsed wrong.

**Comment by ax3l** [comment]
I consider this change too trivial (1 line per file in avg) to be bent on being mentioned in the header (if not already) with an other line change per file.

I updated them manually to avoid missing or breaking some.

**Comment by PrometheusPi** [comment]
There is a some cases of `catch` **without** `const` references remaining:
 - [`ArgsParser.cpp`](https://github.com/ComputationalRadiationPhysics/picongpu/blob/499fc5195e927ec53259f9a1ae661b67f842d682/src/picongpu/ArgsParser.cpp#L123) 
 - [`MySimulation.hpp`](https://github.com/ComputationalRadiationPhysics/picongpu/blob/499fc5195e927ec53259f9a1ae661b67f842d682/src/picongpu/include/simulationControl/MySimulation.hpp#L586) (is `const`, but syntax confuses)

**Comment by ax3l** [comment]
yes I kept them since they are known to work. `const` would probably still be more excellent.

File: src/tools/splash2txt/splash2txt.cpp (hunk 1 above)
Comment (PrometheusPi) [inline src/tools/splash2txt/splash2txt.cpp]: :+1:
